src/app/api/youtube/models.ts

```
@@ -0,0 +1,39 @@
+import { YOUTUBE_CHANNEL_PLAYLISTS } from "~/utils/server-helper";
+
+// TODO: Limit no of recurive calls that could be made, probably limit result to 500, i.e; 10
+async function getAllPlaylists(channelId: string, data: any, playlists: any[]) {
+  const pageToken = data.nextPageToken;
+  if (pageToken) {
+    const response = await fetch(
+      YOUTUBE_CHANNEL_PLAYLISTS(channelId, pageToken)
+    );
+    const result = await response.json();
+    playlists = [...playlists, ...result.items];
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)